Character.h

```diff
@@ -10,65 +10,73 @@
 struct Character
 {
     Character(int hp, int armor_, int attackDamage_ );
+
     virtual ~Character() { }
     
     /*
-     a pure virtual getName function.
-     derived class stores the name, not the base class.
-     */
+    a pure virtual getName function.
+    derived class stores the name, not the base class.
+    */
     
     virtual const std::string& getName() = 0;
+
     virtual std::string getStats() = 0;
     
     virtual void attack( Character& other );
+
     void defend();
     
     void help( Character& other );
     
     int takeDamage(int damage);
     
-    int getHP() const { return hitPoints; }
-    int getArmorLevel() const { return armor; }
-    int getAttackDamage() const { return attackDamage; }
-    bool getIsDefending() const { return isDefending; }
-    
-    const std::vector<std::unique_ptr<Item>>& getHelpfulItems() const { return helpfulItems; }
-    const std::vector<std::unique_ptr<Item>>& getDefensiveItems() const { return defensiveItems; }
+    int getHP() const;
+
+    int getArmorLevel() const;
+
+    int getAttackDamage() const;
+
+    bool getIsDefending() const;
+
+    const std::vector<std::unique_ptr<Item>>& getHelpfulItems() const;
+
+    const std::vector<std::unique_ptr<Item>>& getDefensiveItems() const;
     
-    void boostArmor( int amount )
-    {
-        armor += amount;
-        std::cout << getName() << "'s armor level has been boosted to " << armor << std::endl;
-    }
+    void boostArmor( int amount );
     
-    void boostHitPoints( int amount )
-    {
-        hitPoints += amount;
-        std::cout << getName() << "'s hit point level has been boosted to " << hitPoints << std::endl;
-    }
+    void boostHitPoints( int amount );
     
-    void boostAttackDamage( int amount )
-    {
-        attackDamage += amount;
-        std::cout << getName() << "'s attack damage level has been boosted to " << attackDamage << std::endl;
-    }
-
-    void printStats()
-    {
-        std::cout << getName() << "'s stats: " << std::endl;
-        std::cout << getStats(); //make your getStats() use a function from the Utility.h
-        
-        std::cout << std::endl;
-        std::cout << std::endl;
-    }
+    void boostAttackDamage( int amount );
+
+    void printStats();
+
 protected:
+
     std::vector<std::unique_ptr<Item>> defensiveItems;
+
     std::vector<std::unique_ptr<Item>> helpfulItems;
-    int hitPoints, armor;
+
+    int hitPoints;
+    
+    int armor;
+
     int attackDamage;
+
     bool isDefending = false;
+
 private:
-    std::unique_ptr<int> initialHitPoints, initialArmorLevel, initialAttackDamage;
+
+    std::unique_ptr<int> initialHitPoints;
+
+    std::unique_ptr<int> initialArmorLevel;
+
+    std::unique_ptr<int> initialAttackDamage;
     
     void attackInternal(Character& other);
-};
\ No newline at end of file
+
+    void levelUpHitPoints();
+
+    void levelUpArmor();
+    
+    void levelUpAttackDamage();    
```

Comment:
None of this needed editing.  Put it back the way it was.
- [ ] revise.

Reply:
done